dhall/GitHubWithSourceRepo-project.dhall

```diff
@@ -0,0 +1,26 @@
+    let GitHubProject : Type = { owner : Text, repo : Text }
+in  let gitHubWithSourceRepoProject =
+            λ(repoData : ./types/SourceRepo.dhall )
+          → λ(github : GitHubProject)
+          →     let gitHubRoot =
+                      "https://github.com/${github.owner}/${github.repo}"
+            
+            in    ./defaults/Package.dhall 
+                ⫽ { name =
+                      github.repo
+                  , bug-reports =
+                      "${gitHubRoot}/issues"
+                  , homepage =
+                      gitHubRoot
+                  , source-repos =
+                      [   repoData
+                        ⫽ { location =
+                              [ gitHubRoot ] : Optional Text
+                          , type =
+                              [ (constructors ./types/RepoType.dhall ).Git {=}
+                              ] : Optional ./types/RepoType.dhall 
```

Comment:
I'm not sure I like this, tbh. The type of `gitHubWithSourceRepoProject` requires you pass in a `location` and a `type` (as `repoData : SourceRepo`), but then `⫽` will immediately obliterate anything you supply (`⫽` is right biased). `GitHubProject` as before just passes in some defaults there which isn't too bad, but in the general case I find this a bit surprising.

Better would be to have `repoData : SourceRepo - { location, type }`, but that's not easily expressed in Dhall.

Reply:
yeah i would like to have that type, whick is clearly the correc one, as you note